src/core/border_router/routing_manager.cpp

```diff
@@ -116,6 +116,22 @@ Error RoutingManager::Init(uint32_t aInfraIfIndex, bool aInfraIfIsRunning)
     return error;
 }
 
+Error RoutingManager::ChangeInfraNetif(uint32_t aInfraIfIndex, bool aInfraIfIsRunning)
+{
+    Error error = kErrorNone;
+    VerifyOrExit(mInfraIf.IsInitialized(), error = kErrorInvalidState);
+    VerifyOrExit(mInfraIf.GetIfIndex() != aInfraIfIndex, error = kErrorNone);
+    if (mInfraIf.IsRunning())
+    {
+        SuccessOrExit(error = mInfraIf.HandleStateChanged(mInfraIf.GetIfIndex(), false));
```

Comment:
We probably want to send RAs with PIO/RIO of zero valid/preferred lifetime. @abtink @librasungirl

Reply:
Yes. I agree we should. `RoutingManager::Stop()` should do this from `SendRouterAdvertisement(kInvalidateAllPrevPrefixes)`

On infra-if change, we should stop the routing manager and restart it again to ensure that all data and state, such as prefixes and routes, that we have discovered from processing RAs on the original infra-if are cleared. 

The call to `HandleStateChanged()` from here should trigger a `Stop()` but not sure if this is a good way as the final RA may not be sent as the `mInfraIf` state is changed.